c7n/resources/efs.py

```diff
@@ -193,3 +194,52 @@ def process(self, resources):
                     LifecyclePolicies=op_map.get(self.data.get('state')))
             except client.exceptions.FileSystemNotFound:
                 continue
+
+
+@ElasticFileSystem.filter_registry.register('lifecycle-policy')
+class LifecyclePolicy(Filter):
+    """Filters efs based on the state of lifecycle policies
+
+    :example:
+
+      .. code-block:: yaml
+
+            policies:
+              - name: efs-filter-lifecycle
+                resource: efs
+                filters:
+                  - type: lifecycle-policy
+                    state: present
+                    value: AFTER_7_DAYS
+
+    """
+    schema = type_schema(
+        'lifecycle-policy',
+        state={'enum': ['present', 'absent']},
+        value={'type': 'string'},
+        required=['state'])
+
+    permissions = ('elasticfilesystem:DescribeLifecycleConfiguration',)
+
+    def process(self, resources, event=None):
+        resources = self.lfc_evaluation(resources)
+        if self.data.get('value'):
+            config = {'TransitionToIA': self.data.get('value')}
+            if self.data.get('state') == 'present':
+                return [r for r in resources if config in r.get('LifecyclePolicies')]
+            return [r for r in resources if config not in r.get('LifecyclePolicies')]
+        else:
+            if self.data.get('state') == 'present':
+                return [r for r in resources if r.get('LifecyclePolicies')]
+            return [r for r in resources if r.get('LifecyclePolicies') == []]
+
+    def lfc_evaluation(self, resources):
```

Comment:
the method name here could use a different name, its not really evaluating, its just fetching/annotating lifecycle